core/cloudflow-flink/src/main/scala/cloudflow/flink/FlinkStreamlet.scala

```diff
@@ -80,33 +80,47 @@ abstract class FlinkStreamlet extends Streamlet[FlinkStreamletContext] with Seri
       streamletDefinition ← StreamletDefinition.read(config)
     } yield {
       val updatedConfig = streamletDefinition.config.withFallback(config)
-      new FlinkStreamletContextImpl(streamletDefinition, createExecutionEnvironment(updatedConfig), updatedConfig)
+      new FlinkStreamletContextImpl(streamletDefinition,
+                                    createExecutionEnvironment(updatedConfig, streamletDefinition.streamletRef),
+                                    updatedConfig)
     }).recoverWith {
       case th ⇒ Failure(new Exception(s"Failed to create context from $config", th))
     }.get
 
+  /**
+   * Populate Flink configuration from Streamlet Flink configuration
+   */
+  private def populateFlinkConfiguration(configuration: Configuration, config: Config): Unit =
```

Comment:
`configs: Config*` and then `configs.forEach` ?

Reply:
Same thing

Reply:
I meant that you can pass in more than one config with each call.

```
populateFlinkConfiguration(configuration, config.atPath(ClusterFlinkJobExecutor.flinkRuntime), config.atPath(ClusterFlinkJobExecutor.streamletRuntimeConfigPath(streamlet)))
```

Reply:
Yea, but I need to preserve the order. Need to make sure that streamlet overwrite runtime. Also, need to know the path to remove it from the key (yesterday's update)

Reply:
It will preserve the order with an array, right_